cmd/kubeadm/app/images/images.go

```diff
@@ -39,10 +40,11 @@ func GetCoreImage(image string, cfg *kubeadmapi.MasterConfiguration, overrideIma
 	}
 	repoPrefix := kubeadmapi.GlobalEnvParams.RepositoryPrefix
 	return map[string]string{
-		KubeEtcdImage:              fmt.Sprintf("%s/%s-%s:%s", repoPrefix, "etcd", runtime.GOARCH, etcdVersion),
+		KubeEtcdImage:              fmt.Sprintf("quay.io/coreos/etcd:%s", cfg.Etcd.Cluster.Version),
```

Comment:
folks will need to be able to overload the registry location, air gap'd installs will have some pre-pull and only allow a secured local registry.